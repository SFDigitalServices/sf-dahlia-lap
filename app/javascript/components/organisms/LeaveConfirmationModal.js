import React from 'react'

import Modal from './Modal'

const LeaveConfirmationModal = ({ isOpen, handleClose, destination }) => (
  <Modal isOpen={isOpen} handleClose={handleClose}>
    <Modal.Body hidden={isOpen} handleClose={handleClose}>
      <Modal.Header id={'leave-confirmation-modal'} title={'Are you sure you want to leave this page?'} />
      <Modal.Content>
        <p>You will lose your unsaved changes.</p>
      </Modal.Content>
      <Modal.Footer>
        <div className='modal-button_item modal-button_primary'>
          <a className='button alert' href={destination}>Discard Changes</a>
        </div>
        <div className='modal-button_item modal-button_secondary'>
          <button className='button no-border' onClick={handleClose} type='button'>Keep Editing</button>
        </div>
      </Modal.Footer>
    </Modal.Body>
  </Modal>
)

export default LeaveConfirmationModal
