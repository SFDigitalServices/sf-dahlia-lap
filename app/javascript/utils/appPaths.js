const toApplicationSupplementals = (applicationId) =>
  `/lease-ups/applications/${applicationId}/supplemental`

const toApplication = (applicationId, showAddBtn = false) => {
  const queryParams = showAddBtn ? '?showAddBtn=true' : ''

  return `/applications/${applicationId}${queryParams}`
}

const toListings = () => '/listings'

// To short form application with lease up headers and tabs
const toLeaseUpShortForm = (applicationId) => `/lease-ups/applications/${applicationId}`

const toApplicationEdit = (applicationId) => `/applications/${applicationId}/edit`

const toApplications = (listingId) => `/listings/${listingId}/applications`

const toApplicationNew = (listingId) => `/listings/${listingId}/applications/new`

const toApplicationsFlagged = (id) => `/applications/flagged/${id}`

const toApplicationsFlaggedIndexBase = () => '/applications/flagged'

const toApplicationsFlaggedIndex = (type) => `${toApplicationsFlaggedIndexBase()}?type=${type}`

const toListingLeaseUps = (listingId) => `/lease-ups/listings/${listingId}`

const toListing = (listingId) => `/listings/${listingId}`

const toLeaseUps = () => '/lease-ups/listings'

const toLeaseUpApplications = (listingId) => `/lease-ups/listings/${listingId}`

const toAttachmentDownload = (fileBaseUrl, file) => {
  switch (file.file_type) {
    case 'Attachment':
      return `${fileBaseUrl}/servlet/servlet.FileDownload?file=${file.id}`
    case 'File':
      return `${fileBaseUrl}/sfc/servlet.shepherd/version/download/${file.id}`
    default:
      return null
  }
}

export default {
  toApplicationSupplementals,
  toApplication,
  toApplications,
  toLeaseUpShortForm,
  toApplicationsFlagged,
  toApplicationsFlaggedIndex,
  toApplicationsFlaggedIndexBase,
  toListingLeaseUps,
  toApplicationEdit,
  toListing,
  toListings,
  toApplicationNew,
  toLeaseUps,
  toLeaseUpApplications,
  toAttachmentDownload
}
