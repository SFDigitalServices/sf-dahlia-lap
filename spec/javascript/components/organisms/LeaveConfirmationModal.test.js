/* global mount */
import React from 'react'

import appPaths from '~/utils/appPaths'
import LeaveConfirmationModal from 'components/organisms/LeaveConfirmationModal'

describe('LeaveConfirmationModal', () => {
  const isOpen = true
  const handleClose = jest.fn()
  const destination = appPaths.toApplication('asdf1234')

  test('it should render successfully', () => {
    const wrapper = mount(
      <LeaveConfirmationModal
        isOpen={isOpen}
        handleClose={handleClose}
        destination={destination} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('it should close when the appropriate buttons are clicked', () => {
    const wrapper = mount(
      <LeaveConfirmationModal
        isOpen={isOpen}
        handleClose={handleClose}
        destination={destination} />
    )

    wrapper.find('.modal-button_secondary button').simulate('click')
    wrapper.find('.close-reveal-modal').simulate('click')

    expect(handleClose.mock.calls).toHaveLength(2)
  })
})
