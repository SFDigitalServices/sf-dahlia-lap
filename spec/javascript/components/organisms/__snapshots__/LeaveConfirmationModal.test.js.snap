// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveConfirmationModal it should render successfully 1`] = `
<LeaveConfirmationModal
  destination="/applications/asdf1234"
  handleClose={[MockFunction]}
  isOpen={true}
>
  <ConfirmationModal
    isOpen={true}
    onCloseClick={[MockFunction]}
    onSecondaryClick={[MockFunction]}
    primaryButtonDestination="/applications/asdf1234"
    primaryButtonIsAlert={true}
    primaryText="Discard Changes"
    secondaryText="Keep Editing"
    subtitle="You will lose your unsaved changes."
    title="Are you sure you want to leave this page?"
    titleId="leave-confirmation-modal"
  >
    <Modal
      handleClose={[MockFunction]}
      isOpen={true}
    >
      <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        isOpen={true}
        onRequestClose={[MockFunction]}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        role="dialog"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "content": Object {
              "border": "0",
              "borderRadius": "4px",
              "bottom": "auto",
              "left": "50%",
              "minHeight": "10rem",
              "opacity": "1",
              "overflow": "visible",
              "padding": "0px",
              "position": "fixed",
              "right": "auto",
              "top": "50%",
              "transform": "translate(-50%,-50%)",
              "width": "600px",
            },
            "overlay": Object {
              "backgroundColor": "rgba(0,0,0,0.3)",
            },
          }
        }
      >
        <Portal
          containerInfo={
            <div
              class="ReactModalPortal"
            >
              <div
                class="ReactModal__Overlay ReactModal__Overlay--after-open"
                style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(0, 0, 0, 0.3);"
              >
                <div
                  class="ReactModal__Content ReactModal__Content--after-open"
                  role="dialog"
                  style="position: fixed; top: 50%; left: 50%; border: 0px; background: rgb(255, 255, 255); overflow: visible; border-radius: 4px; outline: none; padding: 0px; opacity: 1; min-height: 10rem; transform: translate(-50%,-50%); width: 600px;"
                  tabindex="-1"
                >
                  <div
                    aria-hidden="true"
                    aria-labelledby="modalTitle"
                    role="dialog"
                  >
                    <header
                      class="modal-inner margin-top"
                      id="leave-confirmation-modal"
                    >
                      <h1
                        class="modal-title t-gamma no-margin"
                      >
                        Are you sure you want to leave this page?
                      </h1>
                    </header>
                    <section
                      class="modal-inner"
                    >
                      <p>
                        You will lose your unsaved changes.
                      </p>
                    </section>
                    <footer
                      class="modal-footer bg-dust"
                    >
                      <div
                        class="modal-button-group row"
                      >
                        <div
                          class="modal-button_item modal-button_secondary"
                        >
                          <a
                            class="button alert"
                            href="/applications/asdf1234"
                            type="button"
                          >
                            Discard Changes
                          </a>
                        </div>
                        <div
                          class="modal-button_item modal-button_secondary"
                        >
                          <button
                            class="button no-border"
                            type="button"
                          >
                            Keep Editing
                          </button>
                        </div>
                      </div>
                    </footer>
                    <button
                      aria-label="Close modal"
                      class="button button-link close-reveal-modal"
                    >
                      <span
                        class="sr-only"
                      >
                        Close
                      </span>
                      <span
                        class="ui-icon ui-medium i-primary"
                      >
                        <svg>
                          <use
                            xlink:href="#i-close"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          }
        >
          <ModalPortal
            ariaHideApp={false}
            bodyOpenClassName="ReactModal__Body--open"
            closeTimeoutMS={0}
            defaultStyles={
              Object {
                "content": Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                },
                "overlay": Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.75)",
                  "bottom": 0,
                  "left": 0,
                  "position": "fixed",
                  "right": 0,
                  "top": 0,
                },
              }
            }
            isOpen={true}
            onRequestClose={[MockFunction]}
            parentSelector={[Function]}
            portalClassName="ReactModalPortal"
            role="dialog"
            shouldCloseOnEsc={true}
            shouldCloseOnOverlayClick={true}
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
            style={
              Object {
                "content": Object {
                  "border": "0",
                  "borderRadius": "4px",
                  "bottom": "auto",
                  "left": "50%",
                  "minHeight": "10rem",
                  "opacity": "1",
                  "overflow": "visible",
                  "padding": "0px",
                  "position": "fixed",
                  "right": "auto",
                  "top": "50%",
                  "transform": "translate(-50%,-50%)",
                  "width": "600px",
                },
                "overlay": Object {
                  "backgroundColor": "rgba(0,0,0,0.3)",
                },
              }
            }
          >
            <div
              className="ReactModal__Overlay ReactModal__Overlay--after-open"
              onClick={[Function]}
              onMouseDown={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(0,0,0,0.3)",
                  "bottom": 0,
                  "left": 0,
                  "position": "fixed",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <div
                className="ReactModal__Content ReactModal__Content--after-open"
                onClick={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                role="dialog"
                style={
                  Object {
                    "WebkitOverflowScrolling": "touch",
                    "background": "#fff",
                    "border": "0",
                    "borderRadius": "4px",
                    "bottom": "auto",
                    "left": "50%",
                    "minHeight": "10rem",
                    "opacity": "1",
                    "outline": "none",
                    "overflow": "visible",
                    "padding": "0px",
                    "position": "fixed",
                    "right": "auto",
                    "top": "50%",
                    "transform": "translate(-50%,-50%)",
                    "width": "600px",
                  }
                }
                tabIndex="-1"
              >
                <Component
                  handleClose={[MockFunction]}
                  hidden={true}
                >
                  <div
                    aria-hidden={true}
                    aria-labelledby="modalTitle"
                    role="dialog"
                  >
                    <Component
                      id="leave-confirmation-modal"
                      title="Are you sure you want to leave this page?"
                    >
                      <header
                        className="modal-inner margin-top"
                        id="leave-confirmation-modal"
                      >
                        <h1
                          className="modal-title t-gamma no-margin"
                        >
                          Are you sure you want to leave this page?
                        </h1>
                      </header>
                    </Component>
                    <Component>
                      <section
                        className="modal-inner"
                      >
                        <p>
                          You will lose your unsaved changes.
                        </p>
                      </section>
                    </Component>
                    <Component>
                      <footer
                        className="modal-footer bg-dust"
                      >
                        <div
                          className="modal-button-group row"
                        >
                          <div
                            className="modal-button_item modal-button_secondary"
                          >
                            <a
                              className="button alert"
                              href="/applications/asdf1234"
                              onClick={[Function]}
                              type="button"
                            >
                              Discard Changes
                            </a>
                          </div>
                          <div
                            className="modal-button_item modal-button_secondary"
                          >
                            <button
                              className="button no-border"
                              onClick={[MockFunction]}
                              type="button"
                            >
                              Keep Editing
                            </button>
                          </div>
                        </div>
                      </footer>
                    </Component>
                    <button
                      aria-label="Close modal"
                      className="button button-link close-reveal-modal"
                      onClick={[MockFunction]}
                    >
                      <span
                        className="sr-only"
                      >
                        Close
                      </span>
                      <span
                        className="ui-icon ui-medium i-primary"
                      >
                        <svg>
                          <use
                            xlinkHref="#i-close"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </Component>
              </div>
            </div>
          </ModalPortal>
        </Portal>
      </Modal>
    </Modal>
  </ConfirmationModal>
</LeaveConfirmationModal>
`;
